### PR TITLE
replace modals by dialogs

### PR DESCRIPTION
**Pull request by 3lbanna**

### CODE REVIEW

File: webapp/webapp.code-workspace [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+	"folders": [
+		{
+			"path": "."
+		}
+	],
+	"settings": {}
+}
```
Comment (pl-buiquang) [inline webapp/webapp.code-workspace]: put *.code-workspace in .gitignore and remove this file

File: webapp/src/components/singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx [hunk 2]
```diff
@@ -32,25 +32,32 @@ function CreateCandidateModal(props: PropType) {
...
-      onActionButtonClick={() => onSave(candidate)}
       title={t("xpansion:newCandidate")}
-      rootStyle={{
-        maxWidth: "80%",
-        maxHeight: "70%",
-        display: "flex",
-        flexFlow: "column nowrap",
-        alignItems: "center",
+      contentProps={{
+        sx: { width: "auto", height: "60vh", p: 2 },
```
Comment (pl-buiquang) [inline webapp/src/components/singlestudy/explore/Xpansion/Candidates/CreateCandidateDialog.tsx]: the size is too big

File: webapp/src/components/singlestudy/HomeView/InformationView/Notes/NodeEditorModal/index.tsx [hunk 3]
```diff
@@ -83,41 +67,31 @@ function NoteEditorModal(props: Props) {
...
-      rootStyle={{
-        width: "600px",
-        display: "flex",
-        flexDirection: "column",
-        justifyContent: "flex-start",
-        alignItems: "center",
-        boxSizing: "border-box",
+      title={t("singlestudy:notes")}
+      contentProps={{
+        sx: { width: "100%", height: "500px", p: 0 },
```
Comment (pl-buiquang) [inline webapp/src/components/singlestudy/HomeView/InformationView/Notes/NodeEditorModal/index.tsx]: set min size (it's too small)

**Review by pl-buiquang — CHANGES_REQUESTED** [review]

File: webapp/src/components/common/dialogs/ReaderDialog/styles.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Box, styled } from "@mui/material";
+
+export const Code = styled(Box)(({ theme }) => ({
```
Comment (skamril) [inline webapp/src/components/common/dialogs/ReaderDialog/styles.ts]: `theme` is not used. Then in the convention we said that we used `styled` only when the component is used multiple time.

**Review by pl-buiquang — APPROVED** [review]

**Pull request merged**